internal/cluster/distribution/eks/eksadapter/node_pool_manager.go

```diff
@@ -94,3 +102,103 @@ func (n nodePoolManager) UpdateNodePool(
 func generateNodePoolStackName(clusterName string, poolName string) string {
 	return "pipeline-eks-nodepool-" + clusterName + "-" + poolName
 }
+
+// ListNodePools is for listing node pools from CloudsetFormation and NodePoolLabelSets
+func (n nodePoolManager) ListNodePools(
+	ctx context.Context,
+	c cluster.Cluster,
+	st eks.SecretStore,
+	dcf kubernetes.DynamicClientFactory,
+) ([]eks.NodePool, error) {
+	// CloudsetFormation
+	sessionFactory := workflow.NewAWSSessionFactory(st)
+	client, err := sessionFactory.New(c.OrganizationID, c.SecretID.String(), c.Location)
+	if err != nil {
+		return nil, err
+	}
+
+	cfClient := cloudformation.New(client)
+	describeStacksInput := cloudformation.DescribeStacksInput{}
+	stacksOutput, err := cfClient.DescribeStacks(&describeStacksInput)
+	if err != nil {
+		return nil, err
+	}
+
+	var relevantStacks []*cloudformation.Stack
+	for _, stack := range stacksOutput.Stacks {
+		// TODO: Later a better filtering method will be needed
+		if strings.HasPrefix(*stack.StackName, "pipeline-eks-nodepool-"+c.Name) {
+			relevantStacks = append(relevantStacks, stack)
+		}
+	}
+
+	stackMap := map[string]*cloudformation.Stack{}
+	for _, stack := range relevantStacks {
+		stackMap[*stack.StackName] = stack
+	}
+
+	// NodePoolLabelSets
+	clusterClient, err := dcf.FromSecret(ctx, c.SecretID.String())
+	if err != nil {
+		return nil, err
+	}
+
+	manager := npls.NewManager(clusterClient, global.Config.Cluster.Namespace)
```

Comment:
Do not use global config unless absolutely necessary. In this case, you can either create a factory for the npls manager, or inject the namespace as a dependency in the node pool manager, but at the very least, leave a TODO to not use global config (it's there for legacy reasons)

Reply:
Fixed in b222234fa4df76dfdef041d7367dbae0a048f662